compaction/compaction_strategy.cc

```diff
@@ -176,13 +176,30 @@ double size_tiered_backlog_tracker::backlog(const compaction_backlog_tracker::on
     return b > 0 ? b : 0;
 }
 
-// FIXME: Should provide strong exception safety guarantees
+// Provides strong exception safety guarantees.
 void size_tiered_backlog_tracker::replace_sstables(const std::vector<sstables::shared_sstable>& old_ssts, const std::vector<sstables::shared_sstable>& new_ssts) {
+    std::vector<sstables::shared_sstable> undo_erase;
+    std::vector<sstables::shared_sstable> undo_insert;
+    undo_erase.reserve(old_ssts.size());
+    undo_insert.reserve(new_ssts.size());
+    auto undo = defer([&] () noexcept {
+        clogger.warn("size_tiered_backlog_tracker: replace_sstables failed. Rolling back...");
+        for (auto& sst : undo_insert) {
+            _total_bytes -= sst->data_size();
+            _all.erase(std::move(sst));
```

Comment:
I wonder if boost::icl::interval_map provides has nothrow erase. Usually erasing doesn't need to allocate, but for interval maps, it needs to invoke merging of nodes.